### PR TITLE
Elpp 2373

### PR DESCRIPTION
**Pull request by Jenniferstrej**

### CODE REVIEW

File: provider/glencoe_check.py [hunk 1]
```diff
@@ -60,6 +62,30 @@ def metadata(msid, settings):
...
+    if '<media content-type="glencoe' in xml_str:
+        return True
+    return False
+
+
+def pad_msid(msid):
+    return str(int(msid)).zfill(5)
+
+
+def check_msid(msid):
```
Comment (giorgiosironi) [inline provider/glencoe_check.py]: nice to see that this module is attracting Glencoe-related functionality

File: activity/activity_CopyGlencoeStillImages.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                for jpg in jpgs:
+                    self.store_file(jpg, article_id)
+
+            self.emit_monitor_event(self.settings, article_id, version, run, self.pretty_name, "end",
+                                    "Finished Copying Glencoe still images to CDN. "
+                                    "Article: " + article_id)
+
+            return activity.activity.ACTIVITY_SUCCESS
+        except Exception as e:
+            self.logger.exception(str(e))
```
Comment (giorgiosironi) [inline activity/activity_CopyGlencoeStillImages.py]: The pattern of the other bot activities suggests that this should be:
``
self.logger.exception("Exception when copying Glencoe still images")
```
as `logger.exception` should already capture message and stack trace of the exception

File: tests/provider/test_glencoe_check.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import unittest
```
Comment (giorgiosironi) [inline tests/provider/test_glencoe_check.py]: nice that unit tests also reflect the growing `glencoe_check` module

File: provider/glencoe_check.py [hunk 4]
```diff
@@ -60,6 +62,30 @@ def metadata(msid, settings):
     
     return resp.json()
 
+
+def jpg_href_values(metadata):
```
Comment (giorgiosironi) [inline provider/glencoe_check.py]: are we doing any renaming of the files to a standard format (and is it even possible)? I don't know what is the final word on how the files should be named and whether the name that comes from the Glencoe API is ok

File: register.py [hunk 5]
```diff
@@ -66,6 +66,7 @@ def start(ENV="dev"):
         print 'got response: \n%s' % json.dumps(response, sort_keys=True, indent=4)
 
     activity_names = []
+    activity_names.append("CopyGlencoeStillImages")
     activity_names.append("VerifyImageServer")
```
Comment (giorgiosironi) [inline register.py]: one thing that is missing is adding this file to the VerifyImageServer checks, if it's not already being picked up. This however can be done in another PR later

File: provider/glencoe_check.py (hunk 4 above)
Comment (Jenniferstrej) [inline provider/glencoe_check.py]: We are not doing any remaining. The last 4 comments in the conversation in the Jira ticket make me understand that having the file name the same as the mp4 one but with .jpg extension is fine and Exeter is treating it as a pattern (see James comment) which makes me think I should just take the file name that Exeter provider in the glencoe metadata.

File: register.py (hunk 5 above)
Comment (Jenniferstrej) [inline register.py]: Right, should it be part of the DoD of this task or a new task?

File: activity/activity_CopyGlencoeStillImages.py (hunk 2 above)
Comment (Jenniferstrej) [inline activity/activity_CopyGlencoeStillImages.py]: I didn't realise. I double checked the documentation and pushed changes.

File: register.py (hunk 5 above)
Comment (giorgiosironi) [inline register.py]: (as agreed in standup, new PR but same task)

File: provider/glencoe_check.py (hunk 4 above)
Comment (giorgiosironi) [inline provider/glencoe_check.py]: as in standup, we should fail fast if the filenames provided by Glencoe are not what we expect. This check is not blocking for this PR, but should be in the DOD

**Comment by gnott** [comment]
Sorry for the late review - is there a test for test_jpg_href_values() glencoe provider, where the metadata is either None or blank or malformed? That complicated looking sequence and lambda function looks like it could use some tests to be sure it is performing as expected.

**Pull request merged**